src/completers/git.zsh

```diff
@@ -88,6 +88,51 @@ _fzf_complete_git() {
         return
     fi
 
+    if [[ $subcommand = checkout ]]; then
+        local prefix_option completing_option
+        local git_options_argument_required=(-b -B --orphan --conflict --pathspec-from-file)
+        local git_options_argument_optional=()
+
+        if completing_option=$(_fzf_complete_git_parse_completing_option "$prefix" "$last_argument" "${(F)git_options_argument_required}" "${(F)git_options_argument_optional}"); then
+            if [[ $completing_option = --* ]]; then
+                prefix_option=$completing_option=
+            else
+                prefix_option=${prefix%%${completing_option[-1]}*}${completing_option[-1]}
+            fi
+        fi
+
+        case $completing_option in
+            -b|-B)
+                return
+                ;;
+
+            --conflict)
+                return
+                ;;
+
+            --pathspec-from-file)
+                return
+                ;;
+
+            *)
+                if [[ -z ${${(Q)${(z)arguments}}[(r)--]} ]]; then
+                    _fzf_complete_git-commits '' $@
```

Comment:
This way the commits are completed in the following situation though the `tree-ish` only comes once. The correct completion function is `_fzf_complete_git-ls-file` in this case.

```zsh
git checkout HEAD **<TAB>
```